Add disputes confirmation screen

mgascam opened the pull request:
Fixes #WOOPMNT-5104

Adds a confirmation screen after successful dispute evidence submission to provide clear feedback and next steps to users.

<img width="1070" alt="Screenshot 2025-07-10 at 09 45 47" src="https://github.com/user-attachments/assets/159301a5-7736-4c53-9bad-8b97b4e6ed4b" />

#### Changes proposed in this Pull Request

- New `ConfirmationScreen` component with success messaging
- Modified evidence form to show confirmation after submission

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* If you don't have any test disputes, please follow [the critical flows document](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#dispute-created-notifications) for creating a dispute (card 4000000000002685) in test mode.
* Navigate to a dispute that needs response and click Challenge Dispute
* Fill out the evidence form
* Click "Submit" and confirm in the dialog
* You should see the new confirmation screen with:
  * Success illustration and messaging
  * "What's next?" section with timeline info
  * Important notice about bank decision
  * Action buttons for navigation

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

mgascam committed:
Initial layout

mgascam committed:
CSS tweaks to follow design specs

mgascam committed:
Use external link

mgascam committed:
Use consistent inline notice

mgascam committed:
Use consistent styles for buttons

mgascam committed:
Fix confirmation screen not rendered on submit

mgascam committed:
Redirect to disputes instead of transactions

mgascam committed:
Add browser confirmation dialog on submit

mgascam committed:
Fix save for later redirect

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10906` or branch name `add/woopmnt-5104-confirmation-screen` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/woopmnt-5104-confirmation-screen) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 680ec7606d9aa7fdf08b119dacf57dcb91af7407
- Build time: 2025-07-15 13:30:43 UTC

_Note: the build is updated when a new commit is pushed to this PR._

mgascam committed:
Extract confirmation screen component

rtio commented on client/disputes/new-evidence/confirmation-screen/style.scss:
```diff
@@ -0,0 +1,188 @@
+}
+
+// Responsive adjustments
+@media ( max-width: 768px ) {
+	.wcpay-dispute-evidence-confirmation {
+		padding: 24px 16px;
+
+		&__title {
+			font-size: 24px;
+		}
+
+		&__actions {
+			flex-direction: column;
+			align-items: center;
+
+			.components-button {
+				width: 100%;
+				max-width: 300px;
+			}
+		}
+	}
+}
```
consider place this inside the class rules

mgascam replied:
Done here [58743bc](https://github.com/Automattic/woocommerce-payments/pull/10906/commits/58743bc9214f3f83bcd356cf8f4d03ef6d78717c) with other refactorings

rtio committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio commented on client/disputes/new-evidence/confirmation-screen/index.tsx:
is there a specific reason you created a folder just for this component?

mgascam replied:
I initially added the confirmation screen into the already existing index file. Then I thought it could be extracted, that's why I created the folder and moved the component and styles inside.

Now that you mention, I believe there's no need to create that folder, specially since it won't be easy to reuse it outside of this flow. So I extracted into its own file instead and place it at the same level as other components. Hopefully it's better this way. This is the commit [58743bc](https://github.com/Automattic/woocommerce-payments/pull/10906/commits/58743bc9214f3f83bcd356cf8f4d03ef6d78717c)

mgascam committed:
Revert move confirmation screen to its own folder

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Implement copy revisions

mgascam committed:
Merge remote-tracking branch 'origin/develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Update copy

mgascam committed:
Add changelog

mgascam committed:
Update illustration

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio commented:
@mgascam  due to this last step you have to update the e2e tests that are expecting the user to be redirected to the dispute detail screen.

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+		font-size: var( --xl, 20px );
+		font-style: normal;
+		font-weight: var( --Medium, 500 );
+		line-height: var( --m, 24px );
```
```suggestion
		@include heading-xl;
```

mgascam replied:
I'm using the mixins in this commit [a1ec548](https://github.com/Automattic/woocommerce-payments/pull/10906/commits/a1ec5484960eb7eed361c4bdfa33238473f51faa) thanks!

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+		font-size: var( --m, 13px );
+		font-style: normal;
+		font-weight: var( --Regular, 400 );
+		line-height: var( --s, 20px );
```
```suggestion
		@include body-md;
```

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+		font-style: normal;
+		font-weight: var( --Medium, 500 );
+		line-height: var( --m, 24px );
+		color: var( --Scales-Grays-gray-900, #1e1e1e );
```
```suggestion
		color: var( --Scales-Grays-gray-900, $gray-900 );
```

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+		font-style: normal;
+		font-weight: var( --Regular, 400 );
+		line-height: var( --s, 20px );
+		color: var( --Scales-Grays-gray-900, #1e1e1e );
```
```suggestion
		color: var( --Scales-Grays-gray-900, $gray-900 );
```

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+			font-style: normal;
+			font-weight: var( --Medium, 500 );
+			line-height: var( --m, 24px );
+			color: var( --Scales-Grays-gray-900, #1e1e1e );
```
```suggestion
			color: var( --Scales-Grays-gray-900, $gray-900 );
```

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+		margin-bottom: 24px;
+
+		h3 {
+			font-size: var( --xl, 20px );
+			font-style: normal;
+			font-weight: var( --Medium, 500 );
+			line-height: var( --m, 24px );
```
```suggestion
			@include heading-xl;
```

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+			li {
+				padding-left: 12px;
+				position: relative;
+				font-size: var( --m, 13px );
+				font-style: normal;
+				font-weight: var( --Regular, 400 );
+				line-height: var( --s, 20px );
```
```suggestion
				@include body-md;
```

rtio commented on client/disputes/new-evidence/style.scss:
```diff
@@ -424,3 +424,134 @@
+				font-style: normal;
+				font-weight: var( --Regular, 400 );
+				line-height: var( --s, 20px );
+				color: var( --Scales-Grays-gray-900, #1e1e1e );
```
```suggestion
				color: var( --Scales-Grays-gray-900, $gray-900 );
```

rtio commented:
on mobile, the buttons below are misaligned
 
<img width="499" height="750" alt="image" src="https://github.com/user-attachments/assets/3a80f2e0-4ee4-466f-8a57-2d96649edc8a" />

I'd suggest following either the approach on dispute detail (my preference) :

<img width="641" height="429" alt="image" src="https://github.com/user-attachments/assets/ba0236cf-5e63-4b5b-9518-74d980b6c065" />

or the navigation:

<img width="497" height="380" alt="image" src="https://github.com/user-attachments/assets/2c4c89eb-4993-4594-a754-29fdb9c4293c" />

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

mgascam committed:
Add ConfirmationScreen tests

mgascam committed:
PR feedback: use CSS typography mixins

mgascam committed:
Fix buttons misaligned in mobile

mgascam commented:
> on mobile, the buttons below are misaligned

Thanks for catching it @rtio! I'm addressing the issue in this commit: [178af4b](https://github.com/Automattic/woocommerce-payments/pull/10906/commits/178af4b94354c03f64fdef1d8586c8516f12d6d9)

Hopefully it looks better now:

<img width="321" height="654" alt="Screenshot 2025-07-10 at 17 25 09" src="https://github.com/user-attachments/assets/9b23bd3f-a764-4ad5-a2a8-ef386a1edafa" />

mgascam commented:
@rtio I addressed your comments so it is ready for a new review, but still need to fix the e2e test.

mgascam committed:
Attempt fix for e2e test

rtio commented:
<!-- linear:replyTo:3035845898 -->

nitpick - I would just invert the order, the button with the accent color should be on top of the outlined one

mgascam committed:
Scroll to top when component mounts

mgascam committed:
Flip the order of actions on smaller screens

mgascam committed:
Change button label to 'Return to transactions'

mgascam committed:
Adjust illustration size

mgascam committed:
Fix test

mgascam commented:
> nitpick - I would just invert the order, the button with the accent color should be on top of the outlined one

This is addressed in [67db366](https://github.com/Automattic/woocommerce-payments/pull/10906/commits/67db366e13da22734f9f293af67a729dbdd3367b). I also addressed the comments after the design review ([ref](https://linear.app/a8c/issue/WOOPMNT-5104/confirmation-screen#comment-97f3d2ae))

mgascam committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio committed:
fix font size and copy

rtio committed:
fix missing var

rtio committed:
last minute request

rtio committed:
fix tests

rtio commented:
@mgascam those are the changes I made here:
- fixed a couple of e2e tests failing
- addressed a last minute request from design
- unrelated to your changes, but I noticed an issue with copy/text size on the `notice` on the bottom of each step so I also fixed it

rtio committed:
Merge branch 'develop' into add/woopmnt-5104-confirmation-screen

rtio commented:
updating branch with the fixes in develop
